### PR TITLE
add docs on git and github

### PR DESCRIPTION
**Pull request by nathanbegbie**
@Rabshab have a quick look at this and see if we need it in addition to existing materials

@salihadfid1 @kevincarolan have created these docs from our sessions yesterday, so you can refer to the commands we learned. Please let me know if this looks helpful.

### CODE REVIEW

File: git_and_github.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+To switch to the master branch
+
+``` powershell
+git checkout master
+```
+
+To switch back to the branch you were just on
+
+``` powershell
+git checkout -
```
Comment (Rabshab) [inline git_and_github.md]: Woah ho. Didn't know that one. Nice wee trick

File: git_and_github.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+A couple of things to check on the pull-request:
+
+* the name of the pull request should describe the work that you've done.e.g.`Added Bayesian Theory to RMarkdown document'.
+* have you assigned someone to review the code?
+* have you assigned yourself to the issue?
+* if this bit of work addresses a particular Issue that was created in Github, please link it in the description by using `#` and the number of the issue.e.g. `Fixes #12` . Github will then link those 2 issues.
+* have you added a helpful description to the pull request so that the person looking at your work understands what you're trying to do.
+
+Once you've created the pull request, and someone has reviewed your code, you can then merge in the code. You do this by pushing the big green button at the bottom of the screen labelled `Create a Merge Commit` .
+
```
Comment (Rabshab) [inline git_and_github.md]: I usually prefer a `squash and merge`. Especially if the work on the branch has been a bit messy. Helps to keep the commits on master nice and clean. Don't feel strongly enough about it to be too prescriptive though so happy to leave this as is.

Comment (nathanbegbie) [reply git_and_github.md]: Good point! Happy to add this in as an option as well though.

Comment (nathanbegbie) [reply git_and_github.md]:
```suggestion
Alternatively, you can also take all of your commits and put them in to a single commit in the main branch by selecting `Squash and Merge`. This helps to keep the commits on master nice and clean, especially if the work on the branch has been a bit messy.
```

**Pull request merged**